### PR TITLE
oneMKL 0.8 version

### PR DESCRIPTION
**Pull request by mkrainiuk**
Reorganization in oneMKL architecture part
Update BLAS domain
Update LAPACK domain

### CODE REVIEW

File: source/elements/oneMKL/source/architecture/api_design.inc.rst [hunk 1]
```diff
@@ -34,18 +35,18 @@ oneMKL uses C++ STL data types for scalars where applicable:
...
-In general, scalar integer arguments to oneMKL routines are 64 bit integers (``std::int64_t`` or ``std::uint64_t``). Integer vectors and matrices may have varying bit widths, defined on a per-routine basis.
+In general, scalar integer arguments to oneMKL routines are 64-bit integers (``std::int64_t`` or ``std::uint64_t``). Integer vectors and matrices may have varying bit widths, defined on a per-routine basis.
 
 .. _onemkl_dpcpp_datatypes:
 
 DPC++ datatype usage
 ++++++++++++++++++++
 
-oneMKL uses following SYCL Language data types and DPC++ Language Extensions data types:
+oneMKL uses following SYCL language data types and DPC++ language extensions data types:
```
Comment (rscohn2) [inline source/elements/oneMKL/source/architecture/api_design.inc.rst:45]: SYCL is not a language and DPC++ does not have language extensions. A feature that is an extension today may be a standard feature tomorrow. Distinguishing between SYCL and DPC++ seems pointless, unless you are committed to supporting a SYCL subset. Keep it simle:

oneMKL uses the following  DPC++  data types:

File: source/elements/oneMKL/source/architecture/api_design.inc.rst [hunk 2]
```diff
@@ -34,18 +35,18 @@ oneMKL uses C++ STL data types for scalars where applicable:
...
 
 * SYCL queue ``sycl::queue`` for scheduling kernels on a SYCL device. See :ref:`onemkl_queues` for more details.
-* SYCL buffer ``sycl::buffer`` for buffer based memory access. See :ref:`onemkl_buffers` for more details.
-* DPC++ Extension Unified Shared Memory (USM) for pointer based memory access. See :ref:`onemkl_usm` for more details.
+* SYCL buffer ``sycl::buffer`` for buffer-based memory access. See :ref:`onemkl_buffers` for more details.
+* DPC++ extension Unified Shared Memory (USM) for pointer-based memory access. See :ref:`onemkl_usm` for more details.
```
Comment (rscohn2) [inline source/elements/oneMKL/source/architecture/api_design.inc.rst:49]: * Unified Shared Memory (USM) for pointer-based memory access. See :ref:`onemkl_usm` for more details.

File: source/elements/oneMKL/source/architecture/architecture.rst [hunk 3]
```diff
@@ -3,8 +3,16 @@
 oneMKL Architecture
 ====================
 
+The oneMKL element of oneAPI has several general assumptions, requirements and recommendations for all domains contained therein.  These will be addressed in this architecture section.  
+In particular, the DPC++ language allows for a great control over the execution of kernels on the various devices.  We discuss the supported execution models of oneMKL APIs in :ref:`onemkl_execution_model`.
```
Comment (rscohn2) [inline source/elements/oneMKL/source/architecture/architecture.rst:7]: > the DPC++ language allows
DPC++ allows

File: source/elements/oneMKL/source/architecture/other_architecture.inc.rst [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+This section covers all other features in the design of oneMKL architecture.
+
+
+
+.. _onemkl_spec_versioning:
+
+oneMKL Specification Versioning Strategy
++++++++++++++++++++++++++++++++++++++++++
+
+This oneMKL specification will use a ``MAJOR.MINOR.PATCH`` approach for its versioning strategy.
```
Comment (rscohn2) [inline source/elements/oneMKL/source/architecture/other_architecture.inc.rst:14]: will use -> uses

File: source/elements/oneMKL/source/architecture/other_architecture.inc.rst [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+
+
+.. _onemkl_spec_versioning:
+
+oneMKL Specification Versioning Strategy
++++++++++++++++++++++++++++++++++++++++++
+
+This oneMKL specification will use a ``MAJOR.MINOR.PATCH`` approach for its versioning strategy.  
+The ``MAJOR`` count will be updated when a significant new feature or domain is added or backward compatibility is broken.
```
Comment (rscohn2) [inline source/elements/oneMKL/source/architecture/other_architecture.inc.rst:15]: will be -> is

File: source/elements/oneMKL/source/architecture/other_architecture.inc.rst [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+
+.. _onemkl_spec_versioning:
+
+oneMKL Specification Versioning Strategy
++++++++++++++++++++++++++++++++++++++++++
+
+This oneMKL specification will use a ``MAJOR.MINOR.PATCH`` approach for its versioning strategy.  
+The ``MAJOR`` count will be updated when a significant new feature or domain is added or backward compatibility is broken.  
+The ``MINOR`` count will be updated when new APIs are changed or added or deleted, or language updated with relatively significant change to the meaning but without breaking backwards compatibility.
```
Comment (rscohn2) [inline source/elements/oneMKL/source/architecture/other_architecture.inc.rst:16]: will be -> is

File: source/elements/oneMKL/source/architecture/other_architecture.inc.rst [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+.. _onemkl_spec_versioning:
+
+oneMKL Specification Versioning Strategy
++++++++++++++++++++++++++++++++++++++++++
+
+This oneMKL specification will use a ``MAJOR.MINOR.PATCH`` approach for its versioning strategy.  
+The ``MAJOR`` count will be updated when a significant new feature or domain is added or backward compatibility is broken.  
+The ``MINOR`` count will be updated when new APIs are changed or added or deleted, or language updated with relatively significant change to the meaning but without breaking backwards compatibility.  
+The ``PATCH`` count will be updated when wording, language or structure is updated without significant change to the meaning or interpretation.
```
Comment (rscohn2) [inline source/elements/oneMKL/source/architecture/other_architecture.inc.rst:17]: will be -> is

File: source/elements/oneMKL/source/architecture/api_design.inc.rst (hunk 1 above)
Comment (mkrainiuk) [reply source/elements/oneMKL/source/architecture/api_design.inc.rst]: Updated

File: source/elements/oneMKL/source/architecture/api_design.inc.rst (hunk 2 above)
Comment (mkrainiuk) [reply source/elements/oneMKL/source/architecture/api_design.inc.rst]: Updated

File: source/elements/oneMKL/source/architecture/architecture.rst (hunk 3 above)
Comment (mkrainiuk) [reply source/elements/oneMKL/source/architecture/architecture.rst]: Updated

File: source/elements/oneMKL/source/architecture/other_architecture.inc.rst (hunk 4 above)
Comment (mkrainiuk) [reply source/elements/oneMKL/source/architecture/other_architecture.inc.rst]: Updated

File: source/elements/oneMKL/source/architecture/other_architecture.inc.rst (hunk 5 above)
Comment (mkrainiuk) [reply source/elements/oneMKL/source/architecture/other_architecture.inc.rst]: Updated

File: source/elements/oneMKL/source/architecture/other_architecture.inc.rst (hunk 6 above)
Comment (mkrainiuk) [reply source/elements/oneMKL/source/architecture/other_architecture.inc.rst]: Updated

File: source/elements/oneMKL/source/architecture/other_architecture.inc.rst (hunk 7 above)
Comment (mkrainiuk) [reply source/elements/oneMKL/source/architecture/other_architecture.inc.rst]: Updated. Thank you!

**Pull request merged**